Azure/azure-linux/ingest/parser.yml

```diff
@@ -0,0 +1,248 @@
+            identify password for \[%{USERNAME:user_name}\]
+  - name: sudo
+    filter: '{{json_event.message.properties.ident == "sudo"}}'
+    external: null
+  - name: grok_omsagent
+    external:
+      name: grok.match
+      properties:
+        input_field: '{{json_event.message.properties.Msg}}'
+        output_field: result
+        pattern: 'omsagent : %{GREEDYDATA:omsagent_message}'
+        custom_patterns: {}
+  - name: omsagent
+    filter: '{{ grok_omsagent.result.omsagent_message != null }}'
+    external:
+      name: kv.parse-kv
+      properties:
+        input_field: '{{ grok_omsagent.result.omsagent_message }}'
+        output_field: result
+        value_sep: '='
+        item_sep: \s;\s
+  - name: grok2_omsagent
+    filter: '{{ grok_omsagent.result.omsagent_message != null }}'
+    external:
+      name: grok.match
+      properties:
+        input_field: '{{omsagent.result.COMMAND }}'
+        output_field: result
+        pattern: '%{UNIXPATH:process_executable}( %{GREEDYDATA:process_args})?'
+        custom_patterns: {}
+  - name: omsagent2
+    filter: '{{ grok_omsagent.result.omsagent_message != null }}'
+    external: null
+  - name: action-outcome
+    filter: >-
+      {{json_event.message.properties.ident == "systemd" or
+      json_event.message.properties.ident == "sudo"}}
+    external: null
+stages:
+  stage2:
+    actions:
+      - set:
+          log.level: '{{json_event.message.level}}'
+          process.pid: '{{json_event.message.properties.pid}}'
+          log.hostname: '{{json_event.message.properties.hostname}}'
+          '@timestamp': '{{json_event.message.time}}'
+          azure_linux.message: '{{json_event.message.properties.Msg}}'
+        name: set
+      - set:
+          os.family: linux
+          os.platform: linux
+        name: set
+      - dictionary:
+          err: error
+          crit: critical
+          info: info
+          warn: warn
+          error: error
+          notice: info
+          critical: critical
+        mapping:
+          log.level: log.level
+        name: translate
+  sshd:
+    actions:
+      - set:
+          source.ip: '{{ grok_sshd.result.source_ip }}'
+          user.name: '{{ grok_sshd.result.user_name }}'
+          source.port: '{{ grok_sshd.result.source_port }}'
+          source.domain: '{{ grok_sshd.result.source_domain }}'
+        name: set
+  systemd:
+    actions:
+      - set:
+          user.name: '{{ grok_systemd.result.user_name }}'
+          action.type: '{{ grok_systemd.result.systemd_observation }}'
+        name: set
+      - set:
+          action.name: systemd:session
+          action.type: open
+        filter: '{{ grok_systemd.result.systemd_observation == "Started Session" }}'
+        name: set
+  crond:
+    actions:
+      - set:
+          user.name: '{{ grok_crond.result.user_name }}'
+          process.args: '{{ grok_crond.result.process_args.split() }}'
```

Comment:
Not sure that is is a good idea to use `split()` here. Consider the following command line for example:

```
/bin/sh -c "while true; do sleep 10 ; done"
```

Here, arguments should be the following thing:

```json
[
  "/bin/sh",
  "-c",
  "while true; do sleep 10 ; done"
]
```

I think that inferring arguments is petty difficult to do. I think that we dropped support for trying to extract args in our Linux parsers.

Reply:
split removed